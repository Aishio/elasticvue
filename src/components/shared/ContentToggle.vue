<template>
  <div class="content-toggle">
    <div v-if="firstActive" ref="first" class="content-toggle__item flex pa-0">
      <div class="grow">
        <slot name="first">first</slot>
      </div>
      <span class="align-self-center content-toggle__last_activator" @click="setLast">
        <slot name="last-activator">Use last</slot>
      </span>
    </div>

    <div v-else ref="last" class="content-toggle__item flex pa-0">
      <div class="grow">
        <slot name="last">last</slot>
      </div>
      <span class="align-self-center content-toggle__first_activator" @click="setFirst">
        <slot name="first-activator">Use first</slot>
      </span>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'content-toggle',
    data () {
      return {
        firstActive: true
      }
    },
    methods: {
      setFirst () {
        this.firstActive = true
        this.$emit('changed', this.firstActive)
      },
      setLast () {
        this.firstActive = false
        this.$emit('changed', this.firstActive)
      }
    }
  }
</script>
