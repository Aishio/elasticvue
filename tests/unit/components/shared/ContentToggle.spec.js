import { mount } from '@vue/test-utils'
import ContentToggle from '@/components/shared/ContentToggle'

describe('components/shared/ContentToggle.vue', () => {
  it('should render correct default contents', () => {
    const wrapper = mount(ContentToggle)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render toggled content', () => {
    const wrapper = mount(ContentToggle)
    let activator = wrapper.find('.content-toggle__last_activator')
    activator.trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render original content', () => {
    const wrapper = mount(ContentToggle)
    let lastActivator = wrapper.find('.content-toggle__last_activator')
    lastActivator.trigger('click')
    let firstActivator = wrapper.find('.content-toggle__first_activator')
    firstActivator.trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })
})
