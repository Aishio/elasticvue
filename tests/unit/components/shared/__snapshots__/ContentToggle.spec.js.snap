// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/ContentToggle.vue should render correct default contents 1`] = `
<div
  class="content-toggle"
>
  <div
    class="content-toggle__item flex pa-0"
  >
    <div
      class="grow"
    >
      first
    </div>
     
    <span
      class="align-self-center content-toggle__last_activator"
    >
      Use last
    </span>
  </div>
</div>
`;

exports[`components/shared/ContentToggle.vue should render original content 1`] = `
<div
  class="content-toggle"
>
  <div
    class="content-toggle__item flex pa-0"
  >
    <div
      class="grow"
    >
      first
    </div>
     
    <span
      class="align-self-center content-toggle__last_activator"
    >
      Use last
    </span>
  </div>
</div>
`;

exports[`components/shared/ContentToggle.vue should render toggled content 1`] = `
<div
  class="content-toggle"
>
  <div
    class="content-toggle__item flex pa-0"
  >
    <div
      class="grow"
    >
      last
    </div>
     
    <span
      class="align-self-center content-toggle__first_activator"
    >
      Use first
    </span>
  </div>
</div>
`;
